### ISSUE TITLE
Ability to hide stereotypes in mermaid

### ISSUE THREAD

**Issue by chardskarth** [mermaid-js/mermaid#6082]
### Proposal

"HIDE_STEREOTYPE()" in C4 diagrams

This is already supported in plantuml. This will be helpful to keep simple diagrams concise.

**Comment by VictorGFM** [comment]
Agree, would be great to have that to simplify system visualization

**Comment by knsv** [comment]
[sisyphus-bot] Thanks for this suggestion, @chardskarth! Adding `HIDE_STEREOTYPE()` support to C4 diagrams is a great idea for keeping diagrams clean and concise. Approved as an enhancement -- this one already has an assignee so it's on the radar.

**Comment by filipsajdak** [comment]
In the opt-in unified C4 renderer (#7842, `c4.useUnifiedRenderer: true`) stereotypes are rendered in a `<span class="c4-type">`, so you can hide them today with CSS:

```css
.c4-type { display: none; }
```

If a first-class option would be preferable, a `c4.hideStereotypes` config flag would be a small addition - happy to put up a PR for it if maintainers want it.

**Comment by filipsajdak** [comment]
Not implemented, and not part of the current C4 work - but the migration makes it cheaper than it used to be, so it is worth recording where it would go.

Today the stereotype (`[Software System]`, `[Container: JavaScript]`, and so on) is assembled with the element label when the shape's text is built. In the open layout work (#8042) that happens in one place, a small shared adapter, rather than being spread through the legacy renderer. A `HIDE_STEREOTYPE()` statement would therefore be a grammar rule plus a flag read at that single point, rather than a change to every draw path.

Two design questions a maintainer would need to settle first:

1. Scope. In C4-PlantUML `HIDE_STEREOTYPE()` is global. Mermaid's C4 dialect has per-element `UpdateElementStyle`, so per-element opt-out is a plausible alternative, and the two are not mutually exclusive.
2. Whether it should be a statement at all, or a config key. Mermaid tends to prefer config for presentation switches, and a config key needs no grammar change - but it does not match the PlantUML syntax people are porting from, which is the reason to want the feature in the first place.

Flagging this rather than picking one, since it is a syntax decision. If a maintainer states a preference I am happy to implement it once the layout slice lands - it is genuinely small at that point.

**Comment by pbrolin47** [comment]
Hi @filipsajdak and @chardskarth,
1. I prefer to have config for controlling presentation of stereotypes.  Even if it's a deviation from PlantUML-syntax, I think the gain is to keep presentation/rendering options similar for different diagram types in Mermaid.
2. The UpdateElementStyle is a good option to override the diagram-global option.

**Comment by filipsajdak** [comment]
Thanks @pbrolin47 - implemented exactly as you specified, opened as #8145.

`c4.showStereotypes` (default true) controls it for the diagram, and `UpdateElementStyle(alias, $showStereotype="false")` overrides it for a single element, in either direction. It covers boundary types as well as element type lines, since both are the same thing to a reader.

@chardskarth - this is the `HIDE_STEREOTYPE()` equivalent you asked for, spelled as configuration rather than as a statement, so it works the way presentation options do elsewhere in Mermaid.

Two things worth recording from building it. `UpdateElementStyle` needed no grammar change - the lexer already turns any `$key="value"` into a named pair - and the same statement reaches boundaries as well as elements, because it resolves its alias against the elements first and then the boundaries.
